src/uct/cuda/cuda_ipc/cuda_ipc_ep.c

```diff
@@ -18,29 +29,217 @@ static UCS_CLASS_INIT_FUNC(uct_cuda_ipc_ep_t, uct_iface_t *tl_iface,
+        CUcontext remote_ptr_ctx;                                       \
+        CUpointer_attribute  attribute;                                 \
+                                                                        \
+        if (key->dev_num == (int) cu_device) {                          \
+            attribute = CU_POINTER_ATTRIBUTE_CONTEXT;                   \
+            cu_ret    = cuPointerGetAttribute((void *) &remote_ptr_ctx, attribute, \
+                                              (CUdeviceptr) remote_addr); \
+            if (CUDA_ERROR_INVALID_VALUE != cu_ret) {                   \
+                /* another process's contex using the same device */    \
+                cu_err = cu_ret;                                        \
+            } else if (cu_ret != CUDA_SUCCESS) {                        \
+                cuGetErrorString(cu_ret, &cu_err_str);                  \
+                ucs_error("cuPointerGetAttribute failed ret:%s", cu_err_str); \
+                return UCS_ERR_IO_ERROR;                                \
+            }                                                           \
+            /* assumes iov is uniformly on device or not */             \
+            cu_ret = cuPointerGetAttribute((void *) &local_ptr_ctx, attribute, \
+                                           (CUdeviceptr) iov[0].buffer); \
+            if (cu_ret != CUDA_SUCCESS) {                               \
+                cuGetErrorString(cu_ret, &cu_err_str);                  \
+                ucs_error("cuPointerGetAttribute failed ret:%s", cu_err_str); \
+                return UCS_ERR_IO_ERROR;                                \
+            }                                                           \
+        }                                                               \
+        if ((CUDA_SUCCESS == cu_err) && (key->dev_num == (int) cu_device) && \
+            (local_ptr_ctx == remote_ptr_ctx)) {                        \
+            mapped_rem_addr = (void *) remote_addr;                     \
+        }                                                               \
+        else {                                                          \
+            /* Is uintptr_t equivalent to uint64_t?  */                 \
+            mapped_rem_base_addr = uct_cuda_ipc_ep_attach_rem_seg(ep, iface, \
+                                                                  key); \
+            offset = (uintptr_t) remote_addr - (uintptr_t) key->d_rem_bptr; \
+            if (offset > key->b_rem_len) {                              \
+                ucs_fatal("Access memory outside memory range attempt\n"); \
+            }                                                           \
+            mapped_rem_addr = (void *) ((uintptr_t) mapped_rem_base_addr + offset); \
+        }                                                               \
+    } while(0);
+
```

Comment:
make inline function instead of such big macro

Reply:
pushed in a change for this.